cmd/restic/global.go

```diff
@@ -365,6 +367,21 @@ func OpenRepository(opts GlobalOptions) (*repository.Repository, error) {
 	})
 
 	s := repository.New(be)
+	var c *cache.Cache
+
+	if !opts.NoCache && opts.CacheAll {
+		c, err = cache.New(opts.Repo, opts.CacheDir, cache.LayoutAll)
```

Comment:
I'm a bit worried about special characters in the repository path, that could cause trouble. I'd suggest to hash the repository path and add an prefix before using it as cache id.

Reply:
Yes, paths are very likely to contain [/:]. They can also contain passwords (cf. #2617).

Reply:
I changed this to use `"repo-"+sha256(opts.Repo)`.
Also added these new cache dirs to the `restic cache` command.